CMakeLists.txt

```diff
@@ -131,13 +131,26 @@ if (DROGON_CXX_STANDARD EQUAL 14)
     # With C++14, use boost to support any and string_view
     message(STATUS "use c++14")
     find_package(Boost 1.61.0 REQUIRED)
-    message(STATUS "boost include dir:" ${Boost_INCLUDE_DIR})
+    message(STATUS "Using Boost filesytem and string_view")
+    message(STATUS "Boost include dir: " ${Boost_INCLUDE_DIR})
+    include_directories(${BOOST_INCLUDE_DIRS})
     target_link_libraries(${PROJECT_NAME} PUBLIC Boost::boost)
     list(APPEND INCLUDE_DIRS_FOR_DYNAMIC_VIEW ${Boost_INCLUDE_DIR})
+    add_definitions(-DUSE_BOOST_FILESYSTEM)
 elseif (DROGON_CXX_STANDARD EQUAL 17)
+    # With C++17, use boost to filesystem::path
     message(STATUS "use c++17")
+    find_package(Boost COMPONENTS filesystem system REQUIRED)
+    message(STATUS "Using Boost filesytem")
+    message(STATUS "Boost include dir: " ${Boost_INCLUDE_DIR})
+    include_directories(${BOOST_INCLUDE_DIRS})
+    message(STATUS "Boost libraries: " ${Boost_LIBRARIES})
+    target_link_libraries(${PROJECT_NAME} PUBLIC Boost::filesystem Boost::system)
+    list(APPEND INCLUDE_DIRS_FOR_DYNAMIC_VIEW ${Boost_INCLUDE_DIR})
+    add_definitions(-DUSE_BOOST_FILESYSTEM)
```

Comment:
Is there a reason behind using boost::filesystem in C++17? AFAIK [std::filesystem](https://en.cppreference.com/w/cpp/filesystem) is a C++17 feature

Reply:
Ah. I see the commit. Let's only use boost::filesystem on Windows and OS X when using C++17. Linux and BSD are not affected.

Reply:
Well, since I couldn't test on unixes with c++17 compiler, I was not sure if they all have std::filesystem::path.
And since OSX didn't have it before 10.15 although the compiler is c++17, I thought that it was safer to check for c++20 on all platforms.

Reply:
I chatted with an-tao. We think it's best to reduce dependencies. And the maintainers can handle compilation error when they pop up. The PR is good as long as it passes the CIs.

Reply:
So, I changed the detection by 1st checking for filesystem and then check for filesystem::path by a compilation test.